hudi-common/src/main/java/org/apache/hudi/common/util/InternalSchemaCache.java

```diff
@@ -135,20 +135,19 @@ private static Option<InternalSchema> getSchemaByReadingCommitFile(long versionI
   /**
    * Get internalSchema and avroSchema for compaction/cluster operation.
    *
-   * @param metaClient current hoodie metaClient
+   * @param metaClient                     current hoodie metaClient
    * @param compactionAndClusteringInstant first instant before current compaction/cluster instant
    * @return (internalSchemaStrOpt, avroSchemaStrOpt) a pair of InternalSchema/avroSchema
    */
   public static Pair<Option<String>, Option<String>> getInternalSchemaAndAvroSchemaForClusteringAndCompaction(HoodieTableMetaClient metaClient, String compactionAndClusteringInstant) {
     // try to load internalSchema to support Schema Evolution
     HoodieTimeline timelineBeforeCurrentCompaction = metaClient.getCommitsAndCompactionTimeline().findInstantsBefore(compactionAndClusteringInstant).filterCompletedInstants();
-    Option<HoodieInstant> lastInstantBeforeCurrentCompaction =  timelineBeforeCurrentCompaction.lastInstant();
+    Option<HoodieInstant> lastInstantBeforeCurrentCompaction = timelineBeforeCurrentCompaction.lastInstant();
     if (lastInstantBeforeCurrentCompaction.isPresent()) {
       // try to find internalSchema
-      byte[] data = timelineBeforeCurrentCompaction.getInstantDetails(lastInstantBeforeCurrentCompaction.get()).get();
       HoodieCommitMetadata metadata;
       try {
-        metadata = metaClient.getCommitMetadataSerDe().deserialize(lastInstantBeforeCurrentCompaction.get(), data, HoodieCommitMetadata.class);
+        metadata = timelineBeforeCurrentCompaction.loadInstantContent(lastInstantBeforeCurrentCompaction.get(), HoodieCommitMetadata.class);
```

Comment:
Similar here on using `readCommitMetadata`.  Let's make sure there is no caller of `InstantContent(instant, clazz)` outside `HoodieTimeline` class.